lib/vmfloaty/abs.rb

```diff
@@ -168,7 +176,7 @@ def self.list(verbose, url, os_filter = nil)
   end
 
   # Retrieve an OS from ABS.
-  def self.retrieve(verbose, os_types, token, url, user, options)
+  def self.retrieve(verbose, os_types, token, url, user, options, _ondemand = nil)
```

Comment:
To avoid these kinds of changes in the future, I think it would be clearer to have an `**opts` param here, and then `_ondemand` can be grabbed from that hash. Then in the future if more options are required, they can be included in the opts hash without having to have a huge list of `option = nil` params added, or adding params that will break other clients with the new option.

Reply:
I'm not opposed to that if we need to add additional options in the future. However, I'm not aware of any additional options we will need to add in the near future so would prefer to keep this change as simple as possible for the time being.

Reply:
I'm going to mark this resolved because I'm making some changes to how we run vmpooler and need the functionality. Please let me know if you feel strongly about this @briancain and I am happy to get it added in the future to replace the additional option.